### PR TITLE
8286772: java/awt/dnd/DropTargetInInternalFrameTest/DropTargetInInternalFrameTest.html times out and fails in Windows

### PR DESCRIPTION
**Pull request by manukumarvs**
This test verifies that dragging from one InternalFrame and a drop target in another InternalFrame functions properly.
This fix moves an unstable closed applet based test to open regression based java test, but updated to be more comprehensive with some stabilisation improvements. The closed test was creating noise in CI as it often fails in Windows platform, especially in Windows 11.

Testing:
1. Tested using mach5 20 times per platform and got all Pass.
2. Tested using mach5 5 times specifically on Windows 11 and got all Pass.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.java.net/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8286772](https://bugs.openjdk.java.net/browse/JDK-8286772): java/awt/dnd/DropTargetInInternalFrameTest/DropTargetInInternalFrameTest.html times out and fails in Windows

### Reviewers
 * [Phil Race](https://openjdk.java.net/census#prr) (@prrace - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8753/head:pull/8753` \
`$ git checkout pull/8753`

Update a local copy of the PR: \
`$ git checkout pull/8753` \
`$ git pull https://git.openjdk.java.net/jdk pull/8753/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8753`

View PR using the GUI difftool: \
`$ git pr show -t 8753`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8753.diff">https://git.openjdk.java.net/jdk/pull/8753.diff</a>

</details>

### CODE REVIEW

**Comment by jayathirthrao** [comment]
I dont see an internal PR to remove the applet based closed test. Also why https://bugs.openjdk.java.net/browse/JDK-8286822 is created as enhancement ?

If this test is failing intermittently how many times we have seen this test fail? Like out of 20 times how many times it will fail on Windows 11? This will help us understand how many times we need to test updated test.

**Comment by manukumarvs** [comment]
> I dont see an internal PR to remove the applet based closed test. Also why https://bugs.openjdk.java.net/browse/JDK-8286822 is created as enhancement ?
> 
> If this test is failing intermittently how many times we have seen this test fail? Like out of 20 times how many times it will fail on Windows 11? This will help us understand how many times we need to test updated test.

I will create an Internal PR to remove the closed test after this PR is integrated(for now I have created a closed bug to track this JDK-8286772). This test was failing 1/20 times in Windows 10, but it was failing almost 100% on Windows 11 before the fix. I have attached the failed mach5 runs also in the bug as a confidential comment.

**Comment by jayathirthrao** [comment]
> > I dont see an internal PR to remove the applet based closed test. Also why https://bugs.openjdk.java.net/browse/JDK-8286822 is created as enhancement ?
> > If this test is failing intermittently how many times we have seen this test fail? Like out of 20 times how many times it will fail on Windows 11? This will help us understand how many times we need to test updated test.
> 
> I will create an Internal PR to remove the closed test after this PR is integrated(for now I have created a closed bug to track this JDK-8286772). This test is added as a new test in open space, that's why JDK-8286822 was created as an enhancement. This test was failing 1/20 times in Windows 10, but it was failing almost 100% on Windows 11 before the fix. I have attached the failed mach5 runs also in the bug as a confidential comment.

We dont create 2 separate JBS ID's for opening up closed test case. Opening up of test case and deleting closed test case should be handled in single JBS bug ID. Opening up a regression test case should not be handled as enhancement.

File: test/jdk/java/awt/dnd/DropTargetInInternalFrameTest.java [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+
+    private static class DropTargetPanel extends JPanel
+            implements DropTargetListener {
+
+        public static final int ENTER_CALLED = 0x1;
+        public static final int OVER_CALLED = 0x2;
+        public static final int DROP_CALLED = 0x4;
+        private final Dimension preferredDimension = new Dimension(200, 100);
+        private final CountDownLatch dropLatch;
+        private volatile int calledMethods = 0;
```
Comment (TejeshR13) [inline test/jdk/java/awt/dnd/DropTargetInInternalFrameTest.java:244]: Any reason for changing calledMethods to volatile?

Comment (manukumarvs) [reply test/jdk/java/awt/dnd/DropTargetInInternalFrameTest.java:240]: In order to get the updated data all the time, as this variable will be updated from EDT and the value will be read from main thread also.

**Comment by manukumarvs** [comment]
> > > I dont see an internal PR to remove the applet based closed test. Also why https://bugs.openjdk.java.net/browse/JDK-8286822 is created as enhancement ?
> > > If this test is failing intermittently how many times we have seen this test fail? Like out of 20 times how many times it will fail on Windows 11? This will help us understand how many times we need to test updated test.
> > 
> > 
> > I will create an Internal PR to remove the closed test after this PR is integrated(for now I have created a closed bug to track this JDK-8286772). This test is added as a new test in open space, that's why JDK-8286822 was created as an enhancement. This test was failing 1/20 times in Windows 10, but it was failing almost 100% on Windows 11 before the fix. I have attached the failed mach5 runs also in the bug as a confidential comment.
> 
> We dont create 2 separate JBS ID's for opening up closed test case. Opening up of test case and deleting closed test case should be handled in single JBS bug ID. Opening up a regression test case should not be handled as enhancement. We add so many of these regression tests as part of normal bugs along with product change.

I have corrected this, now handling everything in a single bug [JDK-8286772](https://bugs.openjdk.java.net/browse/JDK-8286772).

**Comment by manukumarvs** [comment]
Any volunteers please?

File: test/jdk/java/awt/dnd/DropTargetInInternalFrameTest.java (hunk 1 above)
Comment (TejeshR13) [reply test/jdk/java/awt/dnd/DropTargetInInternalFrameTest.java:240]: I meant was it causing an issue in the previous test (before updating the test)?

Comment (manukumarvs) [reply test/jdk/java/awt/dnd/DropTargetInInternalFrameTest.java:240]: I think this was not causing any issues on the previous test.

**Review by prrace — APPROVED** [review]

**Pull request closed without merging**